modules/editable-layers/src/edit-modes/delete-mode.ts

```diff
@@ -0,0 +1,27 @@
+import {FeatureCollection} from '../utils/geojson-types';
+
+import {GeoJsonEditMode} from './geojson-edit-mode';
+import {ClickEvent, ModeProps} from './types';
+import {ImmutableFeatureCollection} from './immutable-feature-collection';
+
+export class DeleteMode extends GeoJsonEditMode {
+  handleClick(_event: ClickEvent, props: ModeProps<FeatureCollection>): void {
+    const selectedFeatureIndexes = props.lastPointerMoveEvent.picks.map((pick) => pick.index);
+    if (selectedFeatureIndexes.length > 0) {
+      const updatedData = new ImmutableFeatureCollection(props.data)
+        // in case of overlapping features, delete only the most recent feature
+        .deleteFeatures([selectedFeatureIndexes[0]])
```

Comment:
Nit: Doesn't it look weird to create an ImmutableFeatureCollection and then immediately proceed to mutate it :) Does this copy the entire collection? I am not up to speed on this code.

Reply:
Good point, I reimplemented this by calling `filter` on the received data instead.